draft-ietf-cose-x509.xml

```diff
@@ -392,6 +415,14 @@ COSE_CertHash = [ hashAlg: (int / tstr), hashValue: bstr ]
         In any event, both the signature and certificate validation MUST be completed successfully before acting on any requests.
       </t>
 
+      <t>
+   	Unless it is known that the CA required proof-of-possession of the subject's private key to issue an end-entity certificate, the end-entity certificate MUST be integrity protected by COSE.  Without proof-of-possession, an attacker can trick the CA to issue an identity-misbinding certificate with someone else's "borrowed" public-key but with a different subject.  A MITM attacker can then perform an identity-misbinding attack by replacing the real end-entity certificate in COSE with such an identity-misbinding certificate.
+      </t>
```

Comment:
This doesn't address the case where a CA correctly and intentionally issued two certs for the same key with different characteristics (e.g., key use, expiration, other extensions) and the attacker swapped them.

Maybe this: "When any field in a certificate beyond the key (e.g., key use, expiration, other extensions) is used in security decisions by the receiver, the COSE header containing or referencing the certificate should be in the protected bucket".

Reply:
I think this point to that COSE integrity protection of the end-entity certificate needs to be MUST.

Reply:
This is silly.  If the CA "correctly" and "intentionally" issued two certificates, and an attacker can swap them, (then anyone else could do so as well, in any every other protocol we have) then the receiver or the protocol is broken.  If this needs to go into the protected header, then we can just put the public key in alone.

Reply:
Agreed that other protocols like CMS don't protect against this possible attack. It is a new protocol attack that was somewhat recently uncovered. When new attacks are found, we still fix them when we can even though we can't fix them everywhere. 

I also agree that it rather obscure and that CA's should do this.

I think my comments here are in line with what Ben said a long time ago and with the text in JWS.

All I want is that these headers can be either protected or not and some encouragement that they be protected. We don't have to agree on the exact attack or document the exact attack.

Reply:
Maybe we can discuss this at the COSE interim today